straxen/plugins/peak_processing.py

```diff
@@ -324,3 +324,85 @@ def find_n_competing(peaks, windows, fraction):
             n_tot[i] = n_left[i] + np.sum(areas[i + 1:right_i] > threshold)
 
         return n_left, n_tot
+
+
+class PeakVetoTagging(strax.Plugin):
+    """
+    Plugin which tags S1 peaks according to  muon and neutron-vetos.
+    Tagging S2s is does not make sense as they occur with a delay.
+    However, we compute for both S1/S2 the time delay to the closest veto
+    region.
+
+        * untagged: 0
+        * neutron-veto: 1
+        * muon-veto: 2
+        * both vetos: 3
+    """
+    __version__ = '0.0.1'
+    depends_on = ('peak_basics', 'veto_regions_nv', 'veto_regions_mv')
+    provides = ('peak_veto_tags')
+    save_when = strax.SaveWhen.TARGET
+
+    dtype = strax.time_fields + \
+            [('veto_tag', np.int8,
+              'Veto tag for S1 peaks. unatagged: 0, nveto: 1, mveto: 2, both: 3'),
+             ('time_to_closest_veto', np.int64)]
+
+    def compute(self, peaks, veto_regions_nv, veto_regions_mv):
+        touching_mv = strax.touching_windows(peaks, veto_regions_mv)
+        touching_nv = strax.touching_windows(peaks, veto_regions_nv)
+
+        tags = np.zeros(len(peaks))
+        tags = tag_peaks(tags, touching_nv, 1)
+        tags = tag_peaks(tags, touching_mv, 2)
+
+        tags[peaks['type'] == 2] = 0
+
+        vetos = np.concatenate([veto_regions_nv, veto_regions_mv])
+        vetos = np.sort(vetos, order='time')
+        dt = get_time_to_closest_veto(peaks, vetos)
+
+        return {'time': peaks['time'],
+                'endtime': strax.endtime(peaks),
+                'veto_tag': tags,
+                'time_to_closest_veto': dt,
+                }
+
+
+@numba.njit(cache=True, nogil=True)
+def tag_peaks(tags, touching_windows, tag_number):
+    pre_tags = np.zeros(len(tags), dtype=np.int8)
+    for start, end in touching_windows:
+        pre_tags[start:end] = tag_number
+    tags += pre_tags
+    return tags
+
+
+def get_time_to_closest_veto(peaks, veto_intervals):
+    """
+    Computes time difference between peak and closest veto interval.
+    """
+    vetos = np.zeros(len(veto_intervals)+2, np.int64)
+    vetos[1:-1] = veto_intervals['time']
+    vetos[-1] = 9223372036854775807  # 64 bit infinity
```

Comment:
perhaps we could re-use https://github.com/XENONnT/straxen/blob/master/straxen/plugins/acqmon_processing.py#L10

Reply:
Sure, but I would put also "infinity" there. @AlexElykov do you see any problem if we replace the 1h in `T_NO_VETO_FOUND` by a 64 bit infinity? 1 h is a bit arbitrary.